src/classes/BDI_DataImportService.cls

```diff
@@ -704,20 +704,54 @@ global with sharing class BDI_DataImportService {
     * from the list of DataImport Fields, which have the form DIObject.developerfieldname in the field's help text property.
     * @param dataImportObjectName the bdi object we care about (Contact1, Contact2, Account1, etc)
     * @param targetObjectName the underlying object the bdi object is (ie, Contact, Account, etc)
-    * @param listStrDIFields the DataImport fields to consider
+    * @param dataImportFields the DataImport fields to consider
     * @return Map<String, String> a map of DataImport fields to underlying object fields (eg, Donation_Amount__c => Amount)
     */
     global static Map<String, String> mapFieldsForDIObject(String dataImportObjectName, String targetObjectName, List<String> dataImportFields) {
 
+        BDISettings__c bdiSettings = BDISettings__c.getOrgDefaults();
+
         Map<String, String> dataImportFieldToTargetField = new Map<String, String>();
+        Map<String, String> bdiFIeldMappingMap = new Map<String, String>();
```

Comment:
bdiFIeldMappingMap -> bdiFieldMappingMap (fix casing on 'i' in field)

Reply:
Actually, we prefer to describe what is in a map in its name. targetFieldsBySource? Although in this case I think we'll need to make it a Map<String, Set<String>> so we can support fields with multiple targets, like the CurrencyIsoCode scenario I was describing in standup. CurrencyIsoCode on Data Import could map to both Opportunity and Payment, and we would want to allow organizations to map them to Account and Contact as well if they chose to.

Reply:
This is a more delicate change, I'm going to modify the map from Map<String, String> to Map<String, Set<String>>, but considering that this changes the how the old and new logic is used while retaining the same core logic for booth cases I'm going to refactor this by isolating the core functionality into a separate method to avoid duplicating a big bunch of code.

Reply:
There is one more casing fix needed to lowercase the 'i' in bdiFIeldMappings.